akka-actor-typed-tests/src/test/scala/akka/actor/typed/SupervisionSpec.scala

```diff
@@ -310,6 +310,27 @@ class SupervisionSpec extends ScalaTestWithActorTestKit("""
       }).onFailure[IllegalArgumentException](strategy)
   }
 
+  class FailingRestartBackoffTestSetup(strategy: SupervisorStrategy) {
+    val probe = TestProbe[AnyRef]("evt")
+
+    def behv =
+      supervise(setup[Command] { context =>
+        // inject some delay on here.
+        Thread.sleep(200)
```

Comment:
Why is this needed? Is it to fill up the mailbox with all `msg` so that they are added to the stash when throw?
Can that be replaced with a CountDownLatch instead to make the test more deterministic?